js_modules/dagit/packages/core/src/asset-graph/AssetNode.tsx

```diff
@@ -26,6 +26,21 @@ import {LiveDataForNode} from './Utils';
 import {ASSET_NODE_NAME_MAX_LENGTH} from './layout';
 import {AssetNodeFragment} from './types/AssetNode.types';
 
+export enum StatusCase {
+  LOADING = 'LOADING',
+  SOURCE_OBSERVING = 'SOURCE_OBSERVING',
+  SOURCE_OBSERVED = 'SOURCE_OBSERVED',
+  SOURCE_NEVER_OBSERVED = 'SOURCE_NEVER_OBSERVED',
+  SOURCE_NO_STATE = 'SOURCE_NO_STATE',
+  MATERIALIZING = 'MATERIALIZING',
+  LATE_OR_FAILED = 'LATE_OR_FAILED',
+  NEVER_MATERIALIZED = 'NEVER_MATERIALIZED',
+  MATERIALIZED = 'MATERIALIZED',
+  PARTITIONS_FAILED = 'PARTITIONS_FAILED',
+  PARTITIONS_MISSING = 'PARTITIONS_MISSING',
+  PARTITIONS_MATERIALIZED = 'PARTITIONS_MATERIALIZED',
+}
```

Comment:
Are these based on something defined in Python? Should it be a GraphQL-based enum?

Reply:
This looks good to me 👋   I like this approach of still co-locating the colors + content + status better than the previous iteration, thank you for changing it! I know it's a little weird to build React content in the helper, but it's /really/ nice that you can look at one spot in the code and see everything related to a particular state.